### ISSUE TITLE
u32testw parser does not output correct VM instruction sequence

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#71]
As mentioned in the comment [here](https://github.com/maticnetwork/miden/pull/62/files#r789429689) current implementation of `u32testw` assembly instruction parser (`parse_u32testw()`) produces an incorrect instruction sequence. We could fix it as suggested in the comment, but there might be a better way.

In general, we want `u32testw` to leave `1` at the top of the stack if all 4 top stack elements are `u32` values and `0` otherwise. Thus, we can test if a given element is a `u32` (leaving `1` at the top of the stack if it is) and then use `AND` operation to combine these values. Thus, we don't need to have `NOT` operations after `EQZ` and then we can replace `OR` operations with `AND`. This should also reduce the cycle count needed for this operation by 4.

So, for example, to test the first element, we'd do:
```Rust
span_ops.push(Operation::Dup3);
span_ops.push(Operation::U32split);
span_ops.push(Operation::Swap);
span_ops.push(Operation::Drop);
span_ops.push(Operation::Eqz);
```

Instead of the current:
```Rust
span_ops.push(Operation::Dup3);
span_ops.push(Operation::U32split);
span_ops.push(Operation::Swap);
span_ops.push(Operation::Drop);
span_ops.push(Operation::Eqz);
span_ops.push(Operation::Not);
```

And for the second, we could do:
```Rust
span_ops.push(Operation::Dup3);
span_ops.push(Operation::U32split);
span_ops.push(Operation::Swap);
span_ops.push(Operation::Drop);
span_ops.push(Operation::Eqz);
span_ops.push(Operation::And);
```

Instead of the current:
```Rust
span_ops.push(Operation::Dup3);
span_ops.push(Operation::U32split);
span_ops.push(Operation::Swap);
span_ops.push(Operation::Drop);
span_ops.push(Operation::Eqz);
span_ops.push(Operation::Not);
span_ops.push(Operation::Or);
```

**Comment by grjte** [comment]
>  current implementation of u32split instruction parser produces an incorrect instruction sequence 
> ...
> In general, we want u32split to leave 1 at the top of the stack if all 4 top stack elements are u32 values and 0 otherwise.

I think it may need to be clarified that this is for the parse_u32testw assembly op (which uses the U32split machine op), and not for the parse_u32split assembly operation

**Comment by bobbinth** [comment]
Ah yes! Thank you! Fixed!

**Comment by bobbinth** [comment]
Fixed by #72 and #91
